.github/workflows/docker.yaml

```diff
@@ -0,0 +1,59 @@
+name: Build Docker Image
+
+on:
+  push:
+    branches: [main, master]
+  pull_request:
+    branches: [main, master]
+
+jobs:
+  docker:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - id: extract_branch
+        name: Extract branch name
+        shell: bash
+        run: echo "branch=${GITHUB_HEAD_REF:-${GITHUB_REF#refs/heads/}}" >> $GITHUB_OUTPUT
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Login to Docker Hub
```

Comment:
Do we want to avoid going via docker hub soon and only go via ghcr? You'll need to arrange a new dockerhub account otherwise, and they will definitely consider Avenir commercial

Reply:
Yes, expecting we can remove this soon. I'll get deployment working off ghcr and then can remove this